xonsh/main.py

```diff
@@ -386,6 +372,20 @@ def func_sig_ttin_ttou(n, f):
             code = sys.stdin.read()
             run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                                 mode='exec')
+        if args.force_interactive:
```

Comment:
This seems like you could end up running a script from a file and *then* entering a command loop. That shouldn't happen, even if it is in interactive mode. The command loop should be entered only if we are in interactive mode and we are not running a script  or a command (`-c`).